src/components/InputField.tsx

```diff
@@ -0,0 +1,71 @@
+import { useContext } from 'react';
+import styled from 'styled-components';
+import useInput from '../hooks/useInput';
+import { COLORS } from '../constants/COLORS';
+import { ItemListContext } from '../contexts/ItemListContext';
+import { IItem } from '../types/types';
+
+const InputField = () => {
+  const { inputText, handleInputChange, resetInputText } = useInput('');
+  const { setItemListHandler } = useContext(ItemListContext);
+
+  const handleFormSubmit = (e: React.SyntheticEvent) => {
```

Comment:
```suggestion
  const handleFormSubmit = (e: React.FormEvent<HTMLFormElement>) => {
```

제가 지난주에 현재님 리뷰에 _`React.SyntheticEvent`라는 것이 있습니다~_ 라고 리뷰를 달아서 그런건지는 몰라도, 많은 분들이 이번 과제에서 이벤트 객체에 `React.SyntheticEvent`를 많이 사용하셨더라구요.

결론부터 말씀드리면 **지양**하는게 좋습니다.
저도 지난주에는 이벤트에 대해 잘 이해하지 못한 상태에서 리뷰를 달다 보니, 저 이벤트를 사용할 수 있다고 말씀을 드렸었네요. 그래서 이번주에는 해당 내용에 대해 조금 더 공부를 해보았습니다.

리액트에서 이벤트 처리를 할 때, 이벤트 핸들러에 전달받는 이벤트 객체는 자바스크립트의 네이티브 이벤트 객체가 아닙니다, `SyntheticEvent` 객체가 전달이 됩니다.

왜 그럴까요? SPA의 특성상 사용자와의 인터랙션이 빈번하게 발생합니다. 많은 이벤트가 발생한다는 의미죠. 하지만 그럴 때 마다 이벤트 객체를 매번 생성하게 된다면 메모리 공간과 오버헤드가 영향을 미칠 수 있겠죠. 그래서 리액트는 이벤트가 발생할 때 마다 `SyntheticEvent` 객체를 재사용하여 (기존에 생성되어있던 객체를 nullify하여) 이벤트를 처리하여 퍼포먼스를 향상시킵니다. (**이벤트 풀링**이라고도 합니다)

그래서 리액트에서는 **어떤 이벤트가 발생하던** 간에, `SyntheticEvent` 객체를 호출하고, 여기에 객체의 속성으로 원래 네이티브 이벤트를 알려주는 방식을 사용합니다. 감이 오시나요? `SyntheticEvent`를 사용하는 것은 결국 어떤 이벤트가 와도 괜찮다고 말하는 것과 같죠.

따라서, 만약 다중 이벤트를 폼에서 처리해야 한다면, 해당하는 이벤트들 끼리 묶어 하나의 커스텀 타입을 만드는 것이 더욱 좋은 방법일 것 같습니다.

[SyntheticEvent](https://ko.reactjs.org/docs/events.html)